### PR TITLE
Nmahendru/qr regression vectors

### PR DESCRIPTION
**Pull request by nmahendru**
Update the regression doc. 
Add steps to generate:
1. valid-qrxxxx test vectors.
2. valid-negative-<testcase> test vectors.

The negative tests do add a paradigm that would be useful going forward
to try to validate more codepaths in CI.

### CODE REVIEW

File: java/gui/src/main/java/com/squareup/subzero/SubzeroGui.java [hunk 1]
```diff
@@ -257,30 +257,35 @@ private void signTxTest() throws Exception {
...
-              } else {
-                  System.out.println("testcase " + filePathInJAR + " : FAIL");
-                  fail_valid++;
-              }
-              continue;
-          }
-          fail_valid++;
-          System.out.println("testcase " + filePathInJAR + " : FAIL");
-          continue;
+        // If the filename for the test vector has a "negative" then it's
```
Comment (nmahendru) [inline java/gui/src/main/java/com/squareup/subzero/SubzeroGui.java:260]: The big diff here is due to formatting change.

The only changes here are:
1. Comment update.
2. ```filePathInJAR.contains("required_fields_not_present") && e.toString().contains("REQUIRED_FIELDS_NOT_PRESENT")```

File: core/src/rpc.c [hunk 2]
```diff
@@ -52,7 +52,7 @@ static Result populate_internal_command(InternalCommandRequest * to){
     goto cleanup;
   }
   if(from.which_command != CommandRequest_SignTx_tag){
-    res = Result_COMMAND_DECODE_FAILED;
+    res = Result_COMMAND_SHOULD_BE_SIGNTX;
```
Comment (nmahendru) [inline core/src/rpc.c:55]: added a specific error code for this condition.

File: java/server/src/main/java/com/squareup/subzero/server/resources/GenerateQrCodeResource.java [hunk 3]
```diff
@@ -144,15 +146,12 @@ public QrCode signTxRequest(SignTxRequest request) throws WriterException {
     builder.setSignTx(signTxBuilder);
 
     byte [] command_bytes = builder.build().toByteArray();
-    //TODO: This will be used later to add another test negative test case.
-    // assign an unexpected protobuf schema to make the decoding fail in subzero core.
-    //Internal.InternalCommandRequest.Builder temp_builder = Internal.InternalCommandRequest.newBuilder();
-    //temp_builder.setInitWallet(Internal.InternalCommandRequest.InitWalletRequest.newBuilder());
-    //command_bytes = temp_builder.build().toByteArray();
-    //Sign with a know dev key. This is for testing only.
+    // Sign with a known dev key. This is for testing only.
```
Comment (nmahendru) [inline java/server/src/main/java/com/squareup/subzero/server/resources/GenerateQrCodeResource.java:149]: Deleting comments that did not really end up help in generating a negative test case.

**Comment by nmahendru** [comment]
all regression tests work on hsm.

File: java/gui/src/main/java/com/squareup/subzero/SubzeroGui.java (hunk 1 above)
Comment (oreparaz) [reply java/gui/src/main/java/com/squareup/subzero/SubzeroGui.java:260]: nit: eventually we'd like to simplify all this logic to something like
```
expected = ...
actual = ...
if (expected != actual) ...
```

**Review by oreparaz — APPROVED** [review]

**Review by syncom — APPROVED** [review]
Cool
